pkg/config/config.go

```diff
@@ -37,38 +37,63 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
+	Core Core `json:core`
```

Comment:
We're now down to 1 member here. Why not move the members of Core here and eliminate a level of nesting?

Reply:
I'll offer a preference, but not a substantial argument: Leaving as-is retains the option of backwards-compatibly adding a new top-level field that is unlike a core configuration domain. For example, if we were to introduce fields that describe the configuration itself (i.e. an embedded TypeMeta), at the config root, they would be interspersed with fields that configure the cluster. I like the flexibility, and imagine users accustomed to managing Kubernetes manifests won't be bothered by nesting that's comparable to the "spec" convention.

Reply:
Good point, I always forget that the version field is on its own, not in the metadata block.